### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by kaizipaul**
### For the User model:
- Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.
### For the Post model:
- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.
### I added unit specs for all the models' methods and validations.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -3,6 +3,14 @@ class User < ApplicationRecord
...
   has_many :comments, foreign_key: 'author_id'
 
+  after_initialize :set_post_counter_default
+
+  def set_post_counter_default
+    self.post_counter = 0 if post_counter.nil?
+  end
+
+  private :set_post_counter_default
+
```
Comment (Mihndim2020) [inline app/models/user.rb:16]: - [ ] As it is a [requirement ](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md)for this project, "`Add the following validations:
For the User model:
Name must not be blank.
PostsCounter must be an integer greater than or equal to zero.`", please kindly add the above validations to the user model.

File: app/models/user.rb [hunk 2]
```diff
@@ -3,6 +3,14 @@ class User < ApplicationRecord
   has_many :likes, foreign_key: 'author_id'
   has_many :comments, foreign_key: 'author_id'
```
Comment (Mihndim2020) [inline app/models/user.rb:5]: - [ ] As it is a [requirement ](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md)for this project, "`For the Post model:
Title must not be blank.
Title must not exceed 250 characters.
CommentsCounter must be an integer greater than or equal to zero.
LikesCounter must be an integer greater than or equal to zero.`", please kindly add the above validations to the post model.

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+RSpec.describe User, type: :model do
+  before :each do
+    @user = User.new(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher')
+  end
+
+  it 'name must not be blank' do
+    @user.name = nil
+    expect(@user).to be_valid
+  end
```
Comment (Mihndim2020) [inline spec/models/user_spec.rb]: - [ ] For this test to pass, you will to update your user.rb model with validation "`Name must not be blank.`"  please make sure the validation is added.

File: spec/models/post_spec.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+RSpec.describe Post, type: :model do
+  before :each do
+    @user = User.new(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher')
+    @post = Post.new(title: 'Hi', text: 'My world', comments_counter: 0, likes_counter: 0)
+  end
+
+  it 'checks title presence' do
+    @post.title = nil
+    expect(@post).to_not be_valid
+  end
```
Comment (Mihndim2020) [inline spec/models/post_spec.rb:12]: - [ ] For this test to pass, you will to update your post.rb model with validation "`Title must not be blank`"  please make sure the validation is added.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi @kaizipaul, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark:  Good job with test coverage
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — APPROVED** [review]
Hi @kaizipaul 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 👍 
* All tests passed ✔️ 
* Good looking readme file 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**